aten/src/ATen/cpu/vec256/vec256_base.h

```diff
@@ -667,11 +667,12 @@ inline Vec256<T> operator^(const Vec256<T>& a, const Vec256<T>& b) {
 
 template<class T, typename Op>
 static inline Vec256<T> bitwise_binary_op(const Vec256<T> &a, const Vec256<T> &b, Op op) {
-  __at_align32__ intmax_t buffer[256 / sizeof(intmax_t)];
-  for (ptrdiff_t i = 0; i < sizeof(buffer) / sizeof(buffer[0]); ++ i) {
-    const intmax_t *i_a_ptr = reinterpret_cast<const intmax_t*>((const T*) a) + i;
-    const intmax_t *i_b_ptr = reinterpret_cast<const intmax_t*>((const T*) b) + i;
-    buffer[i] = op(*i_a_ptr, *i_b_ptr);
+  static constexpr auto element_no = 256 / sizeof(intmax_t);
+  __at_align32__ intmax_t buffer[element_no];
+  const intmax_t *a_ptr = reinterpret_cast<const intmax_t*>((const T*) a);
+  const intmax_t *b_ptr = reinterpret_cast<const intmax_t*>((const T*) b);
+  for (unsigned i = 0U; i < element_no; ++ i) {
```

Comment:
Overall looks good. Just one small comment: Could you make both `i` and `element_no` `size_t` instead of `unsigned`?

Reply:
@xuhdev, sure, but can I clarify why? `element_no` should already be of type `size_t`, but if, for some reason, compiler will fail to unroll the loop, iterating over `size_t`(which is 64-bit value on most platforms) might be tad slower than iterating over `unsigned`(which is a 32-bit value)

Reply:
Just for clarity. If you are worried about the size of the integers, actually both `i` and `element_no` can be made `int32_t` (or even `int8_t`) without problem in this context. Using `unsigned` is a bit unexpected though, since it is neither the convention for indexing nor guaranteed to be low-bit (which potentially helps).

Reply:
@xuhdev I'm worried that using 64-bit integer(i.e. size_t) as an iterator might be slightly slower on some platforms than `unsigned int`(which is 32-bit on all currently supported platforms)

Reply:
I meant it's just for clarity. If you meant to use 32-bit integers, `int32_t` would be cleaner IMO.